lib/Display/display.cpp

```diff
@@ -259,100 +247,167 @@ void display::displayHighscore()
-    tft.drawRect(0, 25, 10, 10, COLOR_DIAMOND_BORDER);
-
-    tft.fillTriangle(0, 50, 30, 70, 0, 90, COLOR_ROCK);         //2
-    tft.drawTriangle(0, 50, 30, 70, 0, 90, COLOR_WHEELS);
-    tft.drawTriangle(1, 51, 29, 70, 1, 89, COLOR_WHEELS);
-
-    tft.fillRect(0, 85, 10, 10, COLOR_DIAMOND);                 //1
-    tft.drawRect(0, 85, 10, 10, COLOR_DIAMOND_BORDER);
-
-    //items right above
-    tft.fillTriangle(280, 0, 320, 0, 320, 40, COLOR_GOLD);      //10
-    tft.drawTriangle(280, 0, 320, 0, 320, 40, ILI9341_ORANGE);
-    tft.drawTriangle(281, 1, 319, 1, 319, 39, ILI9341_ORANGE);
-
-    tft.fillRect(250, -2, 40, 20, COLOR_ROCK);                   //9
-    tft.drawRect(250, -2, 40, 20, COLOR_WHEELS);
-
-    tft.fillTriangle(220, 0, 240, 30, 260, 0, COLOR_GOLD);      //8
-    tft.drawTriangle(220, 0, 240, 30, 260, 0, ILI9341_ORANGE);
-    tft.drawTriangle(219, 1, 240, 29, 259, 1, ILI9341_ORANGE);
-
-    tft.fillRect(215, 0, 10, 10, COLOR_DIAMOND);                //7
-    tft.drawRect(215, 0, 10, 10, COLOR_DIAMOND_BORDER);
-
-    tft.fillRect(290, 60, 30, 30, COLOR_GOLD);                  //13
-    tft.drawRect(290, 60, 30, 30, ILI9341_ORANGE);
-    tft.drawRect(291, 61, 29, 29, ILI9341_ORANGE);
-
-    tft.fillTriangle(320, 30, 290, 50, 320, 70, COLOR_ROCK);    //12
-    tft.drawTriangle(320, 30, 290, 50, 320, 70, COLOR_WHEELS);
-    tft.drawTriangle(319, 31, 291, 50, 319, 69, COLOR_WHEELS);
-
-    tft.fillRect(310, 30, 10, 10, COLOR_DIAMOND);               //11
-    tft.drawRect(310, 30, 10, 10, COLOR_DIAMOND_BORDER);
-
-    tft.fillTriangle(320, 90, 280, 110, 320, 130, COLOR_ROCK);  //15
-    tft.drawTriangle(320, 90, 280, 110, 320, 130, COLOR_WHEELS);
-    tft.drawTriangle(319, 91, 281, 110, 319, 129, COLOR_WHEELS);
-
-    tft.fillRect(300, 80, 20, 20, COLOR_GOLD);                  //14
-    tft.drawRect(300, 80, 20, 20, ILI9341_ORANGE);
-    tft.drawRect(301, 81, 19, 19, ILI9341_ORANGE);
-
-    //items right below
-    tft.fillRect(290, 160, 30, 30, COLOR_GOLD);                 //17
-    tft.drawRect(290, 160, 30, 30, ILI9341_ORANGE);
-    tft.drawRect(291, 161, 28, 28, ILI9341_ORANGE);
+    if (material == "Gold")
+    {
+        tft.fillRect(x, y, width, height, COLOR_GOLD);
+        tft.drawRect(x, y, width, height, ILI9341_ORANGE);
+        tft.drawRect(x + 1, y + 1, width - 2, height - 2, ILI9341_ORANGE);
+    }
+    if (material == "Stone")
+    {
+        tft.fillRect(x, y, width, height, COLOR_ROCK);
+        tft.drawRect(x, y, width, height, COLOR_WHEELS);
+    }
+    if (material == "Diamond")
+    {
+        tft.fillRect(x, y, width, height, COLOR_DIAMOND);
+        tft.drawRect(x, y, width, height, COLOR_DIAMOND_BORDER);
+    }
+}
+void display::displayDecorativeTriangle(int x1, String smallerX1, int y1, String smallerY1, int x2, String smallerX2, int y2, String smallerY2,
+                                        int x3, String smallerX3, int y3, String smallerY3, String GoldOfStone)
```

Comment:
inplaats van plus of min mee geven kun je het ook bepalen op basis van de coordinaten. dan heb je het altijd goed zonder er overna te denken.

Reply:
Ty, zal er naar kijken.